codegenerator/cli/src/cli_args/interactive_init/evm_prompts.rs

```diff
@@ -74,38 +78,64 @@ impl ContractImportArgs {
         ))
     }
 
-    #[async_recursion]
-    async fn get_contract_import_selection_from_etherscan(
-        network_with_explorer: &NetworkWithExplorer,
+    async fn get_selected_contract(
+        &self,
+        network: &NetworkWithExplorer,
         contract_address: Address,
-    ) -> Result<SelectedContract> {
-        match SelectedContract::from_etherscan(network_with_explorer, contract_address.clone())
-            .await
-        {
+    ) -> anyhow::Result<SelectedContract> {
+        let result = match contract_import(network, &contract_address, 0).await {
+            Ok(ContractImportResult::Contract(contract_data)) => Ok(contract_data),
+            Ok(ContractImportResult::NotVerified) => {
+                println!("Failed to find the verified contract on a block explorer.");
+                Err(())
```

Comment:
Can you rather return the string in the Err() and just use one `println!` before the next prompt ?

Just my preference to limit side effects where possible and it only made sense to me why you were printing a couple of lines down from here.